pkg/services/helpers/neofs/neofs.go

```diff
@@ -244,31 +245,63 @@ func parseRange(s string) (*object.Range, error) {
 	return r, nil
 }
 
-// ObjectSearch returns a list of object IDs from the provided container.
-func ObjectSearch(ctx context.Context, c Client, priv *keys.PrivateKey, containerIDStr string, prm client.PrmObjectSearch) ([]oid.ID, error) {
-	var (
-		s           = user.NewAutoIDSignerRFC6979(priv.PrivateKey)
-		objectIDs   []oid.ID
-		containerID cid.ID
-	)
-	err := containerID.DecodeString(containerIDStr)
-	if err != nil {
-		return nil, fmt.Errorf("%w: %w", ErrInvalidContainer, err)
-	}
-	reader, err := c.ObjectSearchInit(ctx, containerID, s, prm)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initiate object search: %w", err)
-	}
-	defer reader.Close()
+// ObjectSearch returns a channel of object search results from the provided container.
+func ObjectSearch(ctx context.Context, c Client, priv *keys.PrivateKey, containerIDStr string, filters object.SearchFilters, attrs []string) (<-chan client.SearchResultItem, <-chan error) {
+	out := make(chan client.SearchResultItem)
+	errChan := make(chan error, 1)
```

Comment:
`errChan` shouldn't be buffered if error channel is properly drained on the caller's side.